server/src/main/java/dev/slimevr/platform/SteamVRBridge.java

```diff
@@ -153,143 +154,181 @@ protected void writeBatteryUpdate(Tracker localTracker) {
+			}
+			case LEFT_ELBOW -> {
+				batteryTrackers
+					.add(
+						TrackerUtils
+							.getNonInternalTrackerForBodyPosition(
+								allTrackers,
+								TrackerPosition.LEFT_LOWER_ARM
+							)
 					);
-				tertiaryTracker = TrackerUtils
-					.getNonInternalTrackerForBodyPosition(
-						allTrackers,
-						TrackerPosition.LEFT_SHOULDER
+				batteryTrackers
+					.add(
+						TrackerUtils
+							.getNonInternalTrackerForBodyPosition(
+								allTrackers,
+								TrackerPosition.LEFT_SHOULDER
+							)
 					);
-				break;
-			case RIGHT_ELBOW:
-				secondaryTracker = TrackerUtils
-					.getNonInternalTrackerForBodyPosition(
-						allTrackers,
-						TrackerPosition.RIGHT_LOWER_ARM
+			}
+			case RIGHT_ELBOW -> {
+				batteryTrackers
+					.add(
+						TrackerUtils
+							.getNonInternalTrackerForBodyPosition(
+								allTrackers,
+								TrackerPosition.RIGHT_LOWER_ARM
+							)
 					);
-				tertiaryTracker = TrackerUtils
-					.getNonInternalTrackerForBodyPosition(
-						allTrackers,
-						TrackerPosition.RIGHT_SHOULDER
+				batteryTrackers
+					.add(
+						TrackerUtils
+							.getNonInternalTrackerForBodyPosition(
+								allTrackers,
+								TrackerPosition.RIGHT_SHOULDER
+							)
 					);
-				break;
+			}
 		}
 
 		// If the battery level of the tracker is lower than lowestLevel, then
 		// the battery level of the tracker position becomes lowestLevel.
 		// Tracker voltage is set if the tracker position has a battery level
-		// lower than lowest level and has a battery voltage (owoTrack devices
-		// do not).
-		if (
-			(primaryTracker != null) && (primaryTracker.getBatteryLevel() != null)
-		) {
-			lowestLevel = primaryTracker.getBatteryLevel();
-
-			if (primaryTracker.getBatteryVoltage() != null) {
-				trackerVoltage = primaryTracker.getBatteryVoltage();
-			}
-		}
-		if (
-			(secondaryTracker != null)
-				&& (secondaryTracker.getBatteryLevel() != null)
-				&& (secondaryTracker.getBatteryLevel() < lowestLevel)
-		) {
-			lowestLevel = secondaryTracker.getBatteryLevel();
-
-			if (secondaryTracker.getBatteryVoltage() != null) {
-				trackerVoltage = secondaryTracker.getBatteryVoltage();
-			}
-		}
-		if (
-			(tertiaryTracker != null)
-				&& (tertiaryTracker.getBatteryLevel() != null)
-				&& (tertiaryTracker.getBatteryLevel() < lowestLevel)
-		) {
-			lowestLevel = tertiaryTracker.getBatteryLevel();
-
-			if (tertiaryTracker.getBatteryVoltage() != null) {
-				trackerVoltage = tertiaryTracker.getBatteryVoltage();
+		// lower than the lowest level and has a battery voltage (owoTrack
+		// devices do not).
+		for (Tracker batteryTracker : batteryTrackers) {
+			if (
+				batteryTracker != null
+					&& batteryTracker.getBatteryLevel() != null
+					&& batteryTracker.getBatteryLevel() < lowestLevel
+			) {
+				lowestLevel = batteryTracker.getBatteryLevel();
+
+				if (batteryTracker.getBatteryVoltage() != null) {
+					trackerVoltage = batteryTracker.getBatteryVoltage();
+				}
```

Comment:
This is mainly just discussion, but this code currently doesn't have any predictable behaviour when it comes to which tracker `trackerVoltage` comes from, so this suggestion just uses the lowest battery tracker's value. There's also the alternative of using the highest tracker voltage for charging detection, so we show if *any* tracker is charging. Thoughs on this?
```suggestion
				if (batteryTracker.getBatteryVoltage() != null) {
					trackerVoltage = batteryTracker.getBatteryVoltage();
				} else {
					trackerVoltage = 0;
				}
```
```suggestion
				if (
					batteryTracker.getBatteryVoltage() != null
						&& batteryTracker.getBatteryVoltage() > trackerVoltage
				) {
					trackerVoltage = batteryTracker.getBatteryVoltage();
				}
```

Reply:
i prefer the first, we need to show only the information on lowest battery

Reply:
Actually my second suggestion doesn't match my intended behaviour either, it'd have to be an entirely separate if statement if we want that, so pls don't do as they one suggests.